vm/src/stdlib/os.rs

```diff
@@ -1191,6 +1191,48 @@ fn os_chmod(
     Ok(())
 }
 
+#[cfg(unix)]
+fn os_execv(
+    path: PyStringRef,
+    argv_list: PyIterable<PyStringRef>,
+    vm: &VirtualMachine,
+) -> PyResult<()> {
+    // TODO: argv_list to only accept tuples and lists
+    // "TypeError: execv() arg 2 must be a tuple or list"
+
+    let path = match ffi::CString::new(path.as_str()) {
+        Ok(s) => s,
+        Err(_) => {
+            return Err(vm.new_value_error("embedded null byte".to_owned()));
+        }
+    };
+
+    if argv_list.iter(vm)?.count() == 0 {
+        return Err(vm.new_value_error("execv() arg 2 must not be empty".to_owned()));
+    }
+
+    for arg in argv_list.iter(vm)? {
+        if arg?.as_str().contains('\0') {
+            return Err(vm.new_value_error("embedded null byte".to_owned()));
+        }
+    }
+
+    let argv_list: Vec<ffi::CString> = argv_list
+        .iter(vm)?
+        .map(|entry| ffi::CString::new(entry.unwrap().as_str()).unwrap())
+        .collect();
```

Comment:
The upper null byte check can be done in below `CString::new()`

Reply:
Addressed.